### ISSUE TITLE
total_loss becomes nan after some time during training with two GPU's. No issues with single GPU.

### ISSUE THREAD

**Issue by vijaygill** [tensorpack/tensorpack#1421]
Hi,

I am facing total_loss resulting in nan after sometime during training using two GPU's. The training on same dataset worked successfully with one GPU.

The dataset is a subset of Google Open Images V5 dataset. The polygons were extracted from mask images (part of dataset) using cv2.findContours function in OpenCV.

There is no change to config.py. The command line used is almost similar to balloons example.

I tried using "BACKBONE.NORM=GN" with the initial weights "COCO-MaskRCNN-R50FPN2xGN.npz" but no joy.

The environment is a docker container with created with following following Dockerfile:

```
FROM tensorflow/tensorflow:1.15.2-gpu-py3

RUN apt-get update --fix-missing

RUN apt-get install -y git libsm6 libxrender-dev python3-pyqt5 qttools5-dev-tools pyqt5-dev-tools sqlite3 libcap-dev

RUN pip3 install ipython ipython-genutils "keras<=2.2.5" numpy "pandas<0.25" sklearn Pillow matplotlib scikit-learn scikit-image opencv-python opencv-contrib-python imutils utils tables requests npyscreen ratelim jsonpickle sqlalchemy imgaug tqdm msgpack msgpack_numpy zmq tabulate psutil scipy Cython python-prctl

RUN pip3 install --force-reinstall opencv-python opencv-contrib-python

RUN pip3 install git+https://github.com/cocodataset/cocoapi.git#subdirectory=PythonAPI

ENV TF_FORCE_GPU_ALLOW_GROWTH true

RUN mkdir /code /images /images_masks /training /data2 /workspace

VOLUME /code
VOLUME /images
VOLUME /images_masks
VOLUME /training
VOLUME /data2
VOLUME /workspace

```

### 1. What you did:

(1) **If you're using examples, what's the command you run:**
I am using FasterRCNN example. I copied the train.py and added code to register my own dataset.
```
python /code/train-tensorpack.py --config \
       DATA.BASEDIR=/training \
       "DATA.VAL=('gillsoft_val',)" \
       "DATA.TRAIN=('gillsoft_train',)" \
       TRAIN.BASE_LR=1e-3 \
       TRAIN.EVAL_PERIOD=0 \
       TRAIN.STEPS_PER_EPOCH=250 \
       "PREPROC.TRAIN_SHORT_EDGE_SIZE=[600,1200]" \
       "PREPROC.MAX_SIZE=1200" \
       TRAIN.CHECKPOINT_PERIOD=2 \
       DATA.NUM_WORKERS=1 \
       --load /training/COCO-MaskRCNN-R50FPN2x.npz \
       --logdir /training/logs/gillsoft
```

(2) **If you're using examples, have you made any changes to the examples? Paste `git status; git diff` here:**
I just changed my copy of train.;py (named train-tensorpack.py as shown in above example) to register my dataset. And also use AutoResumeTrainConfig to allow an interruped training. Following output of "sdiff -s" with the original file should give the idea of what I changed.

```
                                                              > import mrcnn_config_global
                                                              >
                                                              | from dataset_tensorpack import register_gillsoft
                                                              >     register_gillsoft(cfg.DATA.BASEDIR)
        logger.set_logger_dir(args.logdir, 'd')               |         logger.set_logger_dir(args.logdir, 'k')
    traincfg = TrainConfig(                                   |     traincfg = AutoResumeTrainConfig(always_resume = True,
        session_init=session_init,                            |         session_init=session_init
        starting_epoch=cfg.TRAIN.STARTING_EPOCH               <

```

(3) **If not using examples, help us reproduce your issue:**

### 2. What you observed:

(1) **Include the ENTIRE logs here:**
Log file attached: [log.log](https://github.com/tensorpack/tensorpack/files/4466344/log.log)

(2) **Other observations, if any:**
![total_cost_nan_issue](https://user-images.githubusercontent.com/8999486/79068977-94ca3180-7cc2-11ea-8eaa-c8dba61a22ca.png)

### 3. What you expected, if not obvious.
Training should work successfully like it did in single GPU.

### 4. Your environment:
Tensorpack is not installed with pip but repository is cloned and checked out with various directories added to python search path.

**Comment by vijaygill** [comment]
I went though other issues related to mine once again and noticed someone had null polygons and I found that 13 instances of masks in my dataset also had null polygons. So I have made changes to my dataset class to ignore such instances.
I have started training again.
I will keep this issue open for some time and if everything goes well, I will close it.

Thanks

**Comment by vijaygill** [comment]
:( No luck. It started getting nan after about 50 epochs.

**Comment by ppwwyyxx** [comment]
As the issue template says we're unable to help much if the model does not converge / work well on __your dataset__. This is not considered an issue of tensorpack.

My guess is you either need to double check that your data format is correct (e.g., open image may have box coordinates in [0, 1]) or you need lower learning rate (use smaller BASE_LR and set WARMUP_INIT_LR=0.0).

**Comment by vijaygill** [comment]
@ppwwyyxx that did not work either. Also even though the Open Images dataset has coordinates in 0.0..1.0 format, I convert them to absolute. Also I mentioned that the same dataset works with training  with single GPU.
Anyway, thanks for your reply.
